src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs

```diff
@@ -499,6 +495,56 @@ private KeyedByTypeCollection<IOperationBehavior> GetIOperationBehaviorAttribute
             }
         }
 
+        // Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+        // that matches in name and parameter types.  Null is returned if there is no match.
+        private static MethodInfo GetCorrespondingMethodFromType(Type type, MethodInfo methodInfo)
+        {
+            Contract.Assert(type != null);
+            Contract.Assert(methodInfo != null);
+
+            if (methodInfo.DeclaringType == type)
+            {
+                return methodInfo;
+            }
+
+            MethodInfo matchingMethod = type.GetTypeInfo().DeclaredMethods.SingleOrDefault(m => MethodsMatch(m, methodInfo));
+            return matchingMethod;
+        }
+
+        // Returns true if the given methods match in name and parameter types
+        private static bool MethodsMatch(MethodInfo method1, MethodInfo method2)
+        {
+            Contract.Assert(method1 != null);
+            Contract.Assert(method2 != null);
+
+            if (method1 == method2)
```

Comment:
I haven't found one.  MethodInfo.Equals and ParameterInfo.Equals all delegate to object.Equals except for generic methods which insist the declaring types are equal.